dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js

```diff
@@ -0,0 +1,302 @@
+                  { isCensored: _.includes(ctrl.censoredStatuses,donor.status) },
+                  donor
+                ),
+                labels: ctrl.tipLabels,
+                unit: 'days'
+              }),
+              placement: 'right',
+              sticky: true
+            });
+          },
+          onMouseLeaveDonor: function () {
+            $scope.$emit('tooltip::hide');
+          },
+          onClickDonor: function (e, donor) {
+            window.open('/donors/'+donor.id, '_blank');
+          }
+        });
+      };
+
+      window.addEventListener('resize', update);
+      update();
+
+      this.isDataSetDisabled = function (dataSet) {
+        return _.includes(ctrl.disabledDataSets, dataSet);
+      };
+
+      this.toggleDataSet = function (dataSet) {
+        ctrl.disabledDataSets = _.xor(ctrl.disabledDataSets, [dataSet]);
+        ctrl.isDataSetDisabled(dataSet);
+        update();
+      };
+
+      this.$onChanges = function (changes) {
+        if (changes.dataSets) {
+          setTimeout(update);
+        }
+      };
+
+      this.$onDestroy = function () {
+        window.removeEventListener('resize', update);
+      };
+
+  }];
+
+  module.component('survivalAnalysisGraph', {
+    templateUrl: '/scripts/survivalanalysis/views/survival-analysis.html',
+    bindings: {
+      dataSets: '<',
+      initialDisabledDataSets: '<',
+      tipLabels: '<',
+      censoredStatuses: '<',
+      palette: '<',
+    },
+    controller: survivalAnalysisController,
+    controllerAs: 'ctrl'
+  });
+
+  function processResponses(responses) {
+    var survivalData = responses.survivalData.plain().results;
+    var setsMeta = responses.setsMeta.plain();
+
+    return survivalData.map(function (dataSet) {
+      var donors = _.flatten(dataSet.intervals.map(function (interval) {
+        return interval.donors.map(function (donor) {
+          return _.extend({}, donor, {
+            survivalEstimate: interval.cumulativeSurvival
+          });
+        });
+      }));
+
+      return {
+        meta: _.find(setsMeta, {id: dataSet.id}),
+        donors: donors
+      };
+    });
+  }
+
+  module
+    .service('SurvivalAnalysisService', [
+      '$q',
+      'Restangular',
+      'SetService',
+      function(
+        $q,
+        Restangular,
+        SetService
+      ) {
+
+      function fetchOverallSurvival (setIds) {
+        var data = setIds;
+        
+
+        var fetchSurvival = Restangular
+          .one('analysis')
+          .post('survival', data, {}, {'Content-Type': 'application/json'})
+          .then(function (response) {
+            return Restangular.one('analysis/survival/' + response.id).get();
+          });
+
+        var fetchSetsMeta = SetService.getMetaData(setIds);
+
+        return $q.all({
+          survivalData: fetchSurvival,
+          setsMeta: fetchSetsMeta,
+        })
+          .then(function (responses) {
+            return processResponses(responses);
+          })
+        ;
```

Comment:
I'm not attached to this but it makes continuations easier to work with
Would prefer semi-colon-free for continuations to be honest :P